backend/accounts/tests/test_customer_workflow.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+from django.urls import reverse
+from rest_framework import status
+from accounts.models import User, CustomerProfile
+
+
+@pytest.mark.django_db
+class TestCustomerWorkflow:
+
+    @pytest.fixture
+    def register_url(self):
+        return reverse("register")
+
+    @pytest.fixture
+    def address_url(self):
+        return reverse("addresses-list")
+
+    @pytest.fixture
+    def customer_user_data(self):
+        return {
+            "email": "mohussien@gmail.com",
+            "first_name": "Mohamed",
+            "last_name": "Hussien",
+            "role": "customer",
+            "password": "mo_Hussien@123",
+            "confirm_password": "mo_Hussien@123",
+        }
+
+    @pytest.fixture
+    def customer_address_data(self):
+        return {
+            "street": "13th Building, Alwaheed, Al Moshier Street",
+            "city": "Alexandria",
+            "governorate": "Alexandria",
+            "postal_code": "21523",
+            "country": "Egypt",
+        }
+
+    def test_register_customer_success(
+        self, client, register_url, customer_user_data
```

Comment:
You’re using client from pytest-django, but since we’re testing DRF endpoints, it’s better to use DRF’s APIClient for better request format handling. But there is no need to change that now.
```suggestion

from rest_framework.test import APIClient

@pytest.fixture
    def client():
        return APIClient()

    def test_register_customer_success(
        self, client, register_url, customer_user_data
```